### PR TITLE
Updated generate.cmd in Resource

### PR DESCRIPTION
**Pull request by luanshixia**
ARM has a new api-version `2017-05-10`. Therefore we updated swagger spec and we need to update here too to reflect the change.

https://github.com/Azure/azure-rest-api-specs/pull/1157

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [x] The `.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/Resource/Management.ResourceManager/generate.cmd
Comment (luanshixia) [inline src/SDKs/Resource/Management.ResourceManager/generate.cmd]: This is a temporary URL. Will update this once swagger change merged.

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

**Comment by shahabhijeet** [comment]
@luanshixia please remove "Do not merge" label once the swagger is in and all the right code is generated and add "needs review" label

File: src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj
Comment (TianoMS) [inline src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj]: >2.0.0-preview [](start = 19, length = 13)

should be 1.6.0-preview.

File: src/SDKs/Resource/Management.ResourceManager/Generated/Models/Page1.cs
Comment (TianoMS) [inline src/SDKs/Resource/Management.ResourceManager/Generated/Models/Page1.cs]: >Page1 [](start = 17, length = 5)

there's an issue in AutoRest. This was generated as Page1. It should be Page.

File: src/SDKs/Resource/Management.ResourceManager/generate.cmd
Comment (TianoMS) [inline src/SDKs/Resource/Management.ResourceManager/generate.cmd]: >resSpecFile [](start = 4, length = 11)

Please run all test cases and make sure they pass.

File: src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj
Comment (luanshixia) [inline src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj]: @TianoMS What about assembly version in AssemblyInfo.cs? Also I noticed other RPs tend to use major version here. Since we have a big change, should we also kick off a new major version?

**Comment by luanshixia** [comment]
@shahabhijeet Please review.

File: src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj
Comment (TianoMS) [inline src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj]: cleared. this should be 2.0.0-preview since we have a breaking change.

**Comment by TianoMS** [comment]
@vivsriaus could you also review this change?

File: src/SDKs/Resource/Resource.Tests/InMemoryTests/DeploymentTests.InMemory.cs
Comment (TianoMS) [inline src/SDKs/Resource/Resource.Tests/InMemoryTests/DeploymentTests.InMemory.cs]: something is missing here. We support parameters for Template and Parameters.

**Comment by shahabhijeet** [comment]
@azuresdkci add to white list

**Comment by shahabhijeet** [comment]
@azuresdkci add to whitelist

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

**Comment by markcowl** [comment]
@azuresdkci add to whitelist

File: src/SDKs/Resource/Management.ResourceManager/generate.cmd
Comment (markcowl) [inline src/SDKs/Resource/Management.ResourceManager/generate.cmd]: Please use the commit id of the last commit for your PR in the rest-api-specs repo for these values - do not refer to specs through the floating 'master' branch. For example: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/blob/2671588431bc88cb1a85899570f6b1bef17ccd11/arm-resources/locks/2016-09-01/swagger/locks.json

**Review by markcowl — CHANGES_REQUESTED** [review]
Please make the requested change to the spec links in generate.cmd.  Otherwise, lgtm

**Review by markcowl — APPROVED** [review]

File: src/SDKs/Resource/Resource.Tests/SessionRecords/ResourceGroups.Tests.LiveDeploymentTests/CreateDeploymentAndValidateProperties.json
Comment (TianoMS) [inline src/SDKs/Resource/Resource.Tests/SessionRecords/ResourceGroups.Tests.LiveDeploymentTests/CreateDeploymentAndValidateProperties.json]: >"FxVersion/4.6.25009.03", [](start = 10, length = 25)

Is this new? We don't seem to have this in the old traces.

Comment (luanshixia) [inline src/SDKs/Resource/Resource.Tests/SessionRecords/ResourceGroups.Tests.LiveDeploymentTests/CreateDeploymentAndValidateProperties.json]: @TianoMS It seems the test framework newly added this user agent.

File: src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj
Comment (shahabhijeet) [inline src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj]: @luanshixia you are still in preview, so no need to update the major version to 2.0.0-preview
you can continue using minor version bump.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj
Comment (shahabhijeet) [inline src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj]: Also update the assembly version and assemblyfile version accordingly.

Comment (ravbhatnagar) [inline src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj]: @luanshixia - Lets change it to 1.6 in that case.

Comment (luanshixia) [inline src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj]: @shahabhijeet @ravbhatnagar @TianoMS Resolved.

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**
